activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -333,7 +333,7 @@ def schema_cache
       # this method must only be called while holding connection pool's mutex
       def expire
         if in_use?
-          if @owner != ActiveSupport::IsolatedExecutionState.context
+          if @owner != ActiveSupport::IsolatedExecutionState.context && @owner.alive?
```

Comment:
I'm a bit on the fence about this one, on one hand it makes sense to ignore a dead owner, as the connection can safely be reassigned.

However it's likely indicative of a deeper issue, some thread didn't release its connection, so that's one thing to probably revert and investigate deeper.